src/lib/comms/sol-oic-security.c

```diff
@@ -53,34 +51,51 @@ struct cred_item {
     /* FIXME: Only symmetric pairwise keys supported at the moment. */
 };
 
-struct creds {
-    struct sol_vector items;
-    const struct sol_oic_security *security;
-};
-
 struct sol_socket *sol_coap_server_get_socket(const struct sol_coap_server *server);
 
 static ssize_t
 creds_get_psk(const void *data, struct sol_str_slice id,
     char *psk, size_t psk_len)
 {
-    const struct creds *creds = data;
-    struct cred_item *iter;
-    uint16_t idx;
+    SOL_BUFFER_DECLARE_STATIC(buf, PATH_MAX);
+    struct sol_cert *cert;
+    struct sol_blob *contents;
+    int r;
 
     SOL_DBG("Looking for PSK with ID=%.*s", (int)id.len, id.data);
 
-    SOL_VECTOR_FOREACH_IDX (&creds->items, iter, idx) {
-        if (sol_str_slice_eq(id, iter->id.slice)) {
-            if (iter->id.slice.len > psk_len)
-                return -ENOBUFS;
+    r = sol_buffer_append_printf(&buf, "%s", OIC_CRED_FILE_PREFIX);
+    SOL_INT_CHECK(r, < 0, -ENOBUFS);
+    r = sol_buffer_append_as_base64(&buf, id, SOL_BASE64_MAP);
```

Comment:
Actually I am using base64 because it is the way ids/psks were persisted in json files. But a "escape for filenames" would solve it here. And I could save the bytes without any encoding to the file too. not a problem.